src/utils/wallet.ts

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+
+import { crypto, HDPublicKey, HDPrivateKey, Address } from 'bitcore-lib';
+import Mnemonic from 'bitcore-mnemonic';
+import { HD_WALLET_ENTROPY } from '../constants';
+import { XPubError } from '../errors';
+import Network from '../models/network';
+import _ from 'lodash';
+
+
+const wallet = {
+  /**
+   * Verify if words passed to generate wallet are valid. In case of invalid, returns message
+   *
+   * @param {string} words Words (separated by space) to generate the HD Wallet seed
+   *
+   * @return {Object} {'valid': boolean, 'message': string, 'words': string} where 'words' is a cleaned
+   * string with the words separated by a single space
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  wordsValid(words: string): {valid: boolean, message: string, words?: string} {
+    let newWordsString = '';
+    if (_.isString(words)) {
+      // 1. Replace all non ascii chars by a single space
+      // 2. Remove one or more spaces (or line breaks) before and after the 24 words
+      // 3. Set text to lower case
+      newWordsString = words.replace(/[^A-Za-z0-9]/g, ' ').replace(/\s+/g, ' ').trim().toLowerCase();
+      const wordsArray = newWordsString.split(' ');
+      if (wordsArray.length !== 24) {
+        // Must have 24 words
+        return {'valid': false, 'message': 'Must have 24 words'};
+      } else if (!Mnemonic.isValid(newWordsString)) {
+        // Check if there is a word that does not belong to the list of possible words
+        const wordlist = Mnemonic.Words.ENGLISH;
+        const errorList: string[] = [];
+
+        for (const word of wordsArray) {
+          if (wordlist.indexOf(word) < 0) {
+            errorList.push(word);
+          }
+        }
+
+        let errorMessage = '';
+        if (errorList.length > 0) {
+          errorMessage = `Invalid words: ${errorList.join(' ')}`;
```

Comment:
I'd return a list of invalid words and let the user decide what to do with it.

Reply:
I just did it but I am thinking about being consistent. For all the other errors I am trowing an exception and letting the user decide what to do. In this specific case I will return `{'valid': false, 'invalidWords': ['walk']}`.

Is it consistent or can we do it better?

Reply:
I feel it is weird to sometimes raise an exception and other times return `{'valid': false}`. I agree that we must be consistent in the way to handle errors. I usually prefer to raise exceptions because it stops the execution if the user does not handle it.

Reply:
I also reviewed the new method and created new discussion points.

Reply:
I've replied in the other discussions